main.tf

```
@@ -117,13 +116,13 @@ resource aws_route "private_routes" {
 }
 
 resource aws_route_table_association "public_route_association" {
-  count          = "${var.az_count}"
+  count          = "${length(var.public_cidr_ranges)}"
```

This line should probably be since we need one association per subnet, the counts should match.
```suggestion
  count          = "${var.az_count * var.public_subnets_per_az}"
```

---

@johnctitus I believe `var.az_count * var.public_subnets_per_az` _should_ always `= length(var.public_cidr_ranges)` anyway, but this would account for the _technically_ valid (albeit weird) case of `<`. Will update.